src/test/java/com/microsoft/aad/msal4j/HttpHeaderTest.java

```diff
@@ -59,4 +69,94 @@ public void testHttpHeaderConstructor_valuesNotSet(){
         Assert.assertNull(httpHeaderMap.get(HttpHeaders.APPLICATION_VERSION_HEADER_NAME));
         Assert.assertNotNull(httpHeaderMap.get(HttpHeaders.CORRELATION_ID_HEADER_NAME));
     }
+
+    @Test
+    public void testHttpHeaderConstructor_userIdentifierUPN() {
+
+        PublicClientApplication app = PublicClientApplication
+                .builder("client-id")
+                .build();
+
+        IAcquireTokenParameters parameters = UserNamePasswordParameters
+                .builder(Collections.singleton("scopes"), "username", "password".toCharArray())
+                .build();
+
+        String upn = "testuser@microsoft.com";
+        RequestContext requestContext = new RequestContext(
+                app,
+                PublicApi.ACQUIRE_TOKEN_BY_USERNAME_PASSWORD,
+                parameters,
+                UserIdentifier.fromUpn(upn));
+
+        HttpHeaders httpHeaders = new HttpHeaders(requestContext);
+
+        Map<String, String> httpHeaderMap = httpHeaders.getReadonlyHeaderMap();
+
+        String expectedValue = String.format("%s:%s", "upn", upn);
```

Comment:
probably better to reuse constant for format from corresponding class

Reply:
Agreed, updated.